src/tests/integration_tests/test_number_citation_validator.py

```diff
@@ -139,13 +134,47 @@
             ),
             [],
         ),
+        (
+            7,
+            (
+                "One Thousand Two Hundred Thirty-Four feet also  567 and there are 56.45 others 123,345",
+                [
+                    {
+                        "Document": "soccer",
+                        "Source": "https://www.sabioholding.com/press-releases/sabio-delivers-11-q2-2023-revenue-growth-led-by-57-increase-in-connected-tv-ott-sales",
+                    }
+                ],
+            ),
+            ["1234", "567", "56.45", "123345"],
+        )(
+            7,
+            "how many dogs are going to be in the rally GGH?",
+            (
+                """
+                In the rally GGH there are going to be One Thousand Two Hundred Thirty-Four dogs. and also one thousand cats. 
+                there are going to be also event for wolves and lions.
+                """,
+                [
+                    {
+                        "Document": "soccer",
+                        "Source": "https://www.sabioholding.com/press-releases/sabio-delivers-11-q2-2023-revenue-growth-led-by-57-increase-in-connected-tv-ott-sales",
+                    }
+                ],
+            ),
+            ["1234"],
+        ),
     ],
 )
 def test_number_citation_succeeds_in_qa(
     get_llm, test_id, input_data, expected_numbers, objective
 ):  # noqa: F811
-    llm = get_llm(
-        __file__, test_number_citation_succeeds_in_qa.__name__ + f"_{str(test_id)}"
+    # llm = get_llm(
+    #     __file__, test_number_citation_succeeds_in_qa.__name__ + f"_{str(test_id)}"
+    # )
+
+    llm = SherpaChatOpenAI(
```

Comment:
Use `get_llm` so that the test results can be cached. When use `get_llm`, you can access the actual openai APIs by adding `--external_api` when running `pytest`